src/main/java/cz/cvut/kbss/termit/service/business/UserService.java

```diff
@@ -307,4 +350,47 @@ public List<RdfsResource> getManagedAssets(@NonNull UserAccount user) {
         return aclService.findAssetsByAgentWithSecurityAccess(user.toUser()).stream()
                          .map(dtoMapper::assetToRdfsResource).collect(Collectors.toList());
     }
+
+    private PasswordChangeRequest createPasswordChangeRequest(UserAccount userAccount) {
+        // delete any existing request for the user
+        passwordChangeRequestRepositoryService.findAllByUserAccount(userAccount)
+                                              .forEach(passwordChangeRequestRepositoryService::remove);
+
+        return passwordChangeRequestRepositoryService.create(userAccount);
+    }
+
+    @Transactional
+    public void requestPasswordReset(String username) {
+        final UserAccount account = repositoryService.findByUsername(username)
+                                               .orElseThrow(() -> NotFoundException.create(UserAccount.class, username));
+        PasswordChangeRequest request = createPasswordChangeRequest(account);
+        passwordChangeNotifier.sendPasswordResetEmail(request);
+    }
+
+    private boolean isValid(PasswordChangeRequest request) {
+        return request.getCreatedAt().plus(securityConfig.getPasswordChangeRequestValidity()).isAfter(Instant.now());
```

Comment:
I suggest using `Utils.timestamp` which truncates the `Instant.now` to millis for easier processing (it would be good to use it for creating PasswordChangeRequests as well).